portia_server/portia_api/resources/spiders.py

```diff
@@ -26,6 +28,23 @@ def get_collection(self):
         return self.project.spiders
 
     @detail_route(methods=['post'])
+    def extract(self, *args, **kwargs):
+        pages = self.page_class(self.data, self.user)
+        try:
+            instance = self.get_instance()
+            spider = load_spider(self.storage, instance)
+        except (ValueError, KeyError, IndexError):
+            import traceback
+            traceback.print_exc()
```

Comment:
Yes. I'm not sure if using all of those errors is too general. We may need to add custom errors when building slybot spiders